### ISSUE TITLE
Translating strings with interpolated components

### ISSUE THREAD

**Issue by arnaudriegert** [gristlabs/grist-core#313]
Say I would like to translate [this line](https://github.com/gristlabs/grist-core/blob/5f66a8f298de5bfc40d7e80361bb80cb72e589b3/app/client/ui/HomeIntro.ts#L105).

A localizable version would look something like:
```ts
t('VisitHelpCenter', { link: helpCenterLink() })
```

and in the translations file:
```json
{
  "VisitHelpCenter": "Visit our {{link}} to learn more."
}
```

However this won't work if `link` is itself a "component" (using React-language here, not sure what the correct vocabulary is with grainjs).

Doing something like:
```ts
`${t('VisitOur')} ${link} ${t('ToLearnMore')}
```
wouldn't work as the order might differ from one language to another.

i18next has solved this for React with the [`<Trans />` component](https://react.i18next.com/latest/trans-component), which is part of react-i18next.

We would need an equivalent of this component that works with grainjs in order to translate those strings.

Do you think this is something that the Grist team could add to the [localization lib](https://github.com/incubateur-territoires/grist-core/blob/06e229beac9ed36826045ee36590bb326f5ea737/app/client/lib/localization.ts)? 🙏🏻 

cc @LouisDelbosc

**Comment by berhalak** [comment]
Hi @arnaudpeich

Thanks for reporting it. You are right, we need something that can do the interpolation on the "components" level. I will try to propose something.

**Comment by arnaudriegert** [comment]
Solved in #321
